src/lib/keto-change-processor.ts

```diff
@@ -30,41 +30,45 @@
 import Config from '../shared/config'
 import { logger } from '../shared/logger'
 
-export class KetoChangeProcessor {
-  queue: string[][];
-  timerOn: boolean;
-  timeoutId: NodeJS.Timeout;
-  updateFn: (subjectObjectCombos: string[]) => Promise<void>;
+interface KetoQueueItem {
+  queueArgs: any;
+  updateFn: (fnArgs: any) => Promise<void>;
+}
+
+export default class KetoChangeProcessor {
+  private static _instance: KetoChangeProcessor;
+  private queue: KetoQueueItem[];
+  private timerOn: boolean;
+  private timeoutId: NodeJS.Timeout;
 
-  constructor (updateFn: (subjectObjectCombos: string[]) => Promise<void>) {
+  private constructor () {
     this.queue = []
     this.timerOn = true
     this.timeoutId = setTimeout(this._processQueue.bind(this))
-    this.updateFn = updateFn
   }
 
-  _pushQueue (subjectObjectCombos: string []) : void {
-    this.queue.push(subjectObjectCombos)
+  private _pushQueue (qItem: KetoQueueItem) : void {
+    this.queue.push(qItem)
   }
 
-  _popQueue () : string[] | undefined {
+  private _popQueue () : KetoQueueItem | undefined {
     return this.queue.shift()
   }
 
-  _getFirstItemInQueue () : string[] | null {
+  private _getFirstItemInQueue () : KetoQueueItem | null {
```

Comment:
Reasonably typical to call this kind of functionality _peek_. The semantic difference being that _get_ might mutate, typically _peek_ does not. Should be slightly clearer to read.
```suggestion
  private _peekFirstItemInQueue () : KetoQueueItem | null {
```